packages/yoastseo/src/languageProcessing/languages/en/helpers/checkIfWordIsComplex.js

```diff
@@ -0,0 +1,42 @@
+import wordComplexity from "../config/internal/wordComplexity";
+
+/**
+ * Checks if a word is complex.
+ *
+ * @param {string} word     The word to check.
+ *
+ * @returns {boolean}    Whether or not a word is complex.
+ */
+export default function checkIfWordIsComplex( word ) {
+	const wordComplexityConfig = wordComplexity;
+	const lengthLimit = wordComplexityConfig.wordLength;
+	const frequencyList = wordComplexityConfig.frequencyList;
+	// Whether uppercased beginning of a word decreases its complexity.
+	const doesUpperCaseDecreaseComplexity = wordComplexityConfig.doesUpperCaseDecreasesComplexity;
+
+	/*
+	 * Check for each word whether it is a complex word or not.
+	 * A word is complex if:
+	 * its length is longer than the limit, AND
+	 * the word is not in the frequency list, AND
+	 * if the word does NOT start with a capital letter
+	 * (for languages that see long words to be less complex if they start with a capital letter)
+	 */
+	if ( word.length <= lengthLimit ) {
+		return false;
+	}
+	if ( frequencyList.includes( word ) ) {
+		return false;
+	}
+
+	if ( doesUpperCaseDecreaseComplexity === true && word[ 0 ].toLowerCase() === word[ 0 ] ) {
```

Comment:
the check for `doesUpperCaseDecreaseComplexity` here is non obvious. In the future it could use a comment explaining it.
I see this was already in the code before and it was just moved. So not part of this PR. But let's look at this in the future.

Reply:
There was already a comment on line 14:
	`// Whether uppercased beginning of a word decreases its complexity.`
	
Let me know if this is not sufficient :)

Reply:
@hdvos I added extra documentation on this file for the detection of English complex words :)